### PR TITLE
chore(gha): replace action for creating github releases

### PR DESCRIPTION
**Pull request by dbyron-sf**
see spinnaker/rosco#966 for background

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -39,16 +39,15 @@ jobs:
           ./gradlew --info -Pversion="${RELEASE_VERSION}" -Pgradle.publish.key="${GRADLE_PUBLISH_KEY}" -Pgradle.publish.secret="${GRADLE_PUBLISH_SECRET}" publishPlugins
       - name: Create release
         if: steps.release_info.outputs.SKIP_RELEASE == 'false'
-        uses: actions/create-release@v1
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        uses: softprops/action-gh-release@v1
```
Comment (jasonmcintosh) [inline .github/workflows/release.yml:42]: softprops?

Comment (dbyron-sf) [reply .github/workflows/release.yml:42]: It's what we're using elsewhere.  It's one of the ones https://github.com/actions/create-release recommends...

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -39,16 +39,15 @@ jobs:
...
         with:
-           tag_name: ${{ github.ref }}
-           release_name: ${{ github.event.repository.name }} ${{ github.ref }}
-           body: |
+          body: |
             ${{ steps.release_info.outputs.CHANGELOG }}
-           draft: false
-           prerelease: ${{ steps.release_info.outputs.IS_CANDIDATE }}
+          draft: false
+          name: ${{ github.event.repository.name }} ${{ github.ref_name }}
```
Comment (jasonmcintosh) [inline .github/workflows/release.yml:47]: Intentional change on the release name from ref to ref_name?  where-as tag name below is still ref?

File: .github/workflows/release.yml (hunk 1 above)
Comment (jasonmcintosh) [reply .github/workflows/release.yml:42]: Looks like a decent GH Releaser shared lib.

File: .github/workflows/release.yml (hunk 2 above)
Comment (dbyron-sf) [reply .github/workflows/release.yml:47]: Yup, otherwise the name of the release is wacky:
image

Comment (jasonmcintosh) [reply .github/workflows/release.yml:47]: ahh nice :) Wasn't sure :)

**Review by jasonmcintosh — APPROVED** [review]

**Pull request merged**